weather-app/frontend/src/components/WeatherDisplay.tsx

```
@@ -0,0 +1,175 @@
+        
+        {error && (
+          <div className="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded mb-4">
+            {error}
+          </div>
+        )}
+        
+        <div className="flex flex-wrap gap-4 justify-center mb-8">
+          <button 
+            onClick={() => setSelectedWeather('auto')}
+            className={`px-4 py-2 rounded ${selectedWeather === 'auto' ? 'bg-blue-500 text-white' : 'bg-gray-200'}`}
+          >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)